src/UI/Components/Diagram/stencil/LeftSidebar.ts

```diff
@@ -103,6 +83,47 @@ export class StencilSidebar {
     this.tabsToolbar.render();
   }
 
+  /**
+   * Toggles the visibility of the tabs by freezing and unfreezing the old and new tabs
+   * and adding and removing the joint-hidden class.
+   *
+   * @private
+   * @param newActiveTab - The new active tab.
+   * @param oldActiveTab - The old active tab.
+   * @param newTabId - The id of the new tab.
+   */
+  private toggleTabVisibility = (newActiveTab: Tab, oldActiveTab: Tab, newTabId: string) => {
+    oldActiveTab.stencil.el.classList.add("joint-hidden");
+    oldActiveTab.stencil.freeze();
+
+    newActiveTab.stencil.el.classList.remove("joint-hidden");
+    newActiveTab.stencil.unfreeze();
+
+    const tabs: NodeListOf<HTMLElement> = document.querySelectorAll(
+      '[aria-label="stencil-sidebar-tabs"] li'
+    );
+
+    tabs.forEach((tab) => tab.classList.toggle("pf-m-current", tab.id === newTabId));
+  };
+
+  /**
+   * Toggles the visibility of the tabs by freezing and unfreezing the old and new tabs
+   * and adding and removing the joint-hidden class.
+   *
+   * @public
+   * @param clickedElement - The clicked element.
+   */
+  public toggleTab = (clickedElement: EventTarget) => {
+    if (clickedElement instanceof HTMLElement) {
+      // The clickedElement can also be the entire tabContainer, which we don't want to react on.
+      clickedElement.innerText === "Inventory" &&
+        this.toggleTabVisibility(this.inventoryTab, this.instanceTab, "inventory-tab");
+
+      clickedElement.innerText === "New" &&
+        this.toggleTabVisibility(this.instanceTab, this.inventoryTab, "new-tab");
+    }
+  };
```

Comment:
Not sure how it is done in the frontend, but couldn't we use an enum to store the possible configurations of this `clickedElement.innerText` instead of hard coding `Inventory/New`?

Reply:
Yes, we do use them too, but I admit I didn't yet focus on that part of the refactoring :D 
I'll update